lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/text/TextLineReader.java

```diff
@@ -43,35 +49,56 @@ public final class TextLineReader
     private boolean closed;
     private long readTimeNanos;
 
-    public TextLineReader(InputStream in, int bufferSize)
+    public static TextLineReader createCompressedReader(InputStream in, int bufferSize, Codec codec)
             throws IOException
     {
-        this(in, bufferSize, 0, Long.MAX_VALUE);
+        CountingInputStream countingInputStream = new CountingInputStream(in);
+        LongSupplier rawInputPositionSupplier = countingInputStream::getCount;
+        in = codec.createStreamDecompressor(countingInputStream);
+        return new TextLineReader(in, bufferSize, 0, OptionalLong.empty(), rawInputPositionSupplier);
     }
 
-    public TextLineReader(InputStream in, int bufferSize, long start, long length)
+    public static TextLineReader createUncompressedReader(InputStream in, int bufferSize)
+            throws IOException
+    {
+        return createUncompressedReader(in, bufferSize, 0, Long.MAX_VALUE);
```

Comment:
Should this delegate to the constructor so that it can pass `OptionalLong.empty()`? Using `Long.MAX_VALUE` seems to have the same problem with overflow as the original code (which seems to be why `OptionalLong` is used here).

Or maybe remove this method since it's only used in tests.